src/TSTransformer/nodes/binding/transformObjectAssignmentPattern.ts

```diff
@@ -38,8 +42,37 @@ export function transformObjectAssignmentPattern(
 				state.prereq(transformInitializer(state, id, property.objectAssignmentInitializer));
 			}
 		} else if (ts.isSpreadAssignment(property)) {
-			DiagnosticService.addDiagnostic(errors.noSpreadDestructuring(property));
-			return;
+			const value = spreadDestructObject(state, parentId, preSpreadNames);
+			const expression = property.expression;
+
+			if (ts.isObjectLiteralExpression(property.expression)) {
+				DiagnosticService.addDiagnostic(errors.noNestedSpreadsInAssignmentPatterns(property));
+				continue;
+			}
+
+			if (
+				ts.isIdentifier(expression) ||
+				ts.isElementAccessExpression(expression) ||
+				ts.isPropertyAccessExpression(expression)
```

Comment:
It looks like the branches for `ts.isElementAccessExpression(expression)` and `ts.isPropertyAccessExpression(expression)` are not covered by unit tests.